recipes/CodeLLM_DevKit/code_summarization.ipynb

```diff
@@ -0,0 +1,426 @@
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# For simplicity, we run the code summarization on a single class and method (this filter can be removed to run this code over the entire application)\n",
+    "target_class = \"org.apache.commons.cli.GnuParser\"\n",
+    "target_method = \"flatten(Options, String[], boolean)\"\n",
+    "\n",
+    "# Iterate over all classes in the application\n",
+    "for class_name in analysis.get_classes():\n",
+    "    if class_name == target_class:\n",
+    "        class_file_path = analysis.get_java_file(qualified_class_name=class_name)\n",
+    "\n",
+    "        # Read code for the class\n",
+    "        with open(class_file_path, \"r\") as f:\n",
+    "            code_body = f.read()\n",
+    "\n",
+    "        # Initialize treesitter utils for the class file content\n",
+    "        tree_sitter_utils = cldk.tree_sitter_utils(source_code=code_body)\n",
+    "                \n",
+    "        # Iterate over all methods in class\n",
+    "        for method in analysis.get_methods_in_class(qualified_class_name=class_name):\n",
+    "            if method == target_method:\n",
+    "        \n",
+    "                # Get all the method details\n",
+    "                method_details = analysis.get_method(qualified_class_name=class_name,\n",
+    "                                                     qualified_method_name=method)\n",
+    "                \n",
+    "                # Sanitize the class for analysis with respect to the target method\n",
+    "                sanitized_class = tree_sitter_utils.sanitize_focal_class(method_details.declaration)\n",
+    "        \n",
+    "                # Format the instruction for the given target method and class\n",
+    "                instruction = format_inst(\n",
+    "                    code=sanitized_class,\n",
+    "                    focal_method=method_details.declaration,\n",
+    "                    focal_class=class_name.split(\".\")[-1],\n",
+    "                    language=\"java\"\n",
+    "                )\n",
+    "    \n",
+    "                print(f\"Instruction:\\n{instruction}\\n\")\n",
+    "                print(f\"Generating code summary ...\\n\")\n",
+    "                \n",
+    "                # Prompt the local model on Ollama\n",
+    "                llm_output = prompt_ollama(message=instruction)\n",
+    "        \n",
+    "                # Print the LLM output\n",
+    "                print(f\"LLM Output:\\n{llm_output}\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.11.9"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
I get no output.

Could you add a final summary markdown cell that explains what people should have seen?

Reply:
Added final markdown showing expected output